### PR TITLE
Autocompaction for supported stores

### PR DESCRIPTION
**Pull request by sideshowcoder**

### CODE REVIEW

**Comment by nickcampbell18** [comment]
👍 LGTM

**Comment by nickcampbell18** [comment]
You'll get there in the end!

File: lib/circuitbox/store.rb [hunk 1]
```diff
@@ -0,0 +1,83 @@
+require "delegate"
+
+class Circuitbox
```
Comment (yarmand) [inline lib/circuitbox/store.rb]: I'd rather see a moneta middleware than a wrapper.

With middleware, it is easy to see in one place all the middleware used and understand the behaviour.
With factory wrapping like here, when you have more than one, understanding final behaviour become difficult.

Comment (sideshowcoder) [inline lib/circuitbox/store.rb]: I think I can rewrite this as Moneta Transformer, hooking dump and load.

**Comment by sideshowcoder** [comment]
@yarmand any chance we could push that trough?

**Comment by yarmand** [comment]
well, given the comment above, I'm not sure.
Trying to find a new maintainer in the services team, I'll let him take this decision.

**Comment by sideshowcoder** [comment]
Nah all good :)

**Comment by matthewshafer** [comment]
The `Circuitbox::MemoryStore` periodically compacts itself when used. Closing this as auto compaction should be something the circuit store handles.

**Pull request closed without merging**